templates/nextjs/src/components/Layout.tsx

```diff
@@ -0,0 +1,135 @@
+import { BN, Provider, Wallet, WalletUnlocked } from "fuels";
+import { createContext, useCallback, useEffect, useState } from "react";
+import { BurnerWallet } from "./BurnerWallet";
+import toast, { Toaster } from "react-hot-toast";
+import { Link } from "./Link";
+import { Button } from "./Button";
+import { NODE_URL } from "@/lib";
+
+const BURNER_WALLET_LOCAL_STORAGE_KEY = "create-fuels-burner-wallet-pk";
+
+export const AppContext = createContext<{
+  faucetWallet?: WalletUnlocked;
+  setFaucetWallet?: (wallet: WalletUnlocked) => void;
+
+  burnerWallet?: WalletUnlocked;
+  setBurnerWallet?: (wallet: WalletUnlocked) => void;
+
+  burnerWalletBalance?: BN;
+  setBurnerWalletBalance?: (balance: BN) => void;
+
+  refreshBurnerWalletBalance?: () => Promise<void>;
+}>({});
+
+export const Layout = ({ children }: { children: React.ReactNode }) => {
+  const [faucetWallet, setFaucetWallet] = useState<WalletUnlocked>();
+
+  const [burnerWallet, setBurnerWallet] = useState<WalletUnlocked>();
+  const [burnerWalletBalance, setBurnerWalletBalance] = useState<BN>();
+
+  const [initialTopupDone, setInitialTopupDone] = useState(false);
+
+  useEffect(() => {
+    (async () => {
```

Comment:
Is the usage of `useEffect` for asynchronous calls intentional to not introduce a library dependency? Because the code could be made simpler with e.g. [`useAsync`](https://github.com/streamich/react-use/blob/master/docs/useAsync.md).

Reply:
Yeah, and some people might not be familiar with that library too.